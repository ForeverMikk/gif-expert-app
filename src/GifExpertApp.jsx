import { useState } from "react";
import AddCategory from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";

const GifExpertApp = () => {

    const [categories, setCategories] = useState(['One Punch']);
    // const apiKey = 'PwoOVKqTZhYP4clsAkwqrZ1ueTYAHGnd';

    const onAddCategory = (newCategory) => {
        if(categories.includes(newCategory)) return; //Si la categoria se repite no te deja ingresarla
        setCategories([newCategory, ...categories]);
    }

    return(
        <>
            <h1>GifExpert App</h1>
            <AddCategory 
                onNewCategory={value => onAddCategory(value)} 
            />
            {categories.map((category) => (
                <GifGrid key={category} category={category}/>
            ))}
        </>
    )
}

export default GifExpertApp;