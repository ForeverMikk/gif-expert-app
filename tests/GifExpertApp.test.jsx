/* eslint-disable no-undef */
import { render } from "@testing-library/react";
import GifExpertApp from "../src/GifExpertApp";

describe('pruebas en el Componente <GifExpertApp />', () => { 
    test('debe hacer match con el snapshot', () => { 

        const { container } = render(<GifExpertApp />);
        expect(container).toMatchSnapshot();


    });
});