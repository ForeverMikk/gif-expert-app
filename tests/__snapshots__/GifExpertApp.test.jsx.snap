// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en el Componente <GifExpertApp /> debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    GifExpert App
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar Gifs"
      type="text"
      value=""
    />
  </form>
  <h4>
    One Punch
  </h4>
  <h2>
    Cargando...
  </h2>
  <ol
    class="card-grid"
  />
</div>
`;
